src/common/decorators/api-transaction-response.ts

```diff
@@ -0,0 +1,45 @@
+/* istanbul ignore file */
+
+import { applyDecorators } from '@nestjs/common';
+import {
+  ApiAcceptedResponse,
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiResponseOptions,
+} from '@nestjs/swagger';
+
+import { NotificationPayloadModel } from '~/common/models/notification-payload-model';
+import { TransactionPayloadResultModel } from '~/common/models/transaction-payload-result.model';
+import { TransactionQueueModel } from '~/common/models/transaction-queue.model';
+
+/**
+ * A helper that functions like `ApiCreatedResponse`, that also adds an `ApiAccepted` response in case "submitWithCallback" is used and `ApiOKResponse` if "offline" mode is used
+ *
+ * @param options - these will be passed to the `ApiCreatedResponse` decorator
+ */
+export function ApiTransactionResponse(
+  options: ApiResponseOptions
+): ReturnType<typeof applyDecorators> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const optionType = (options as any).type;
+
+  const extendsQueueModel = optionType.prototype instanceof TransactionQueueModel;
```

Comment:
This instanceof check made for a circular dependency, which moving to its own file resolved.